### PR TITLE
CORE-16237: Unit tests for VerificationRpcRequestProcessor

### PR DESCRIPTION
**Pull request by emilybowe**
Adding unit testing for the new VerificationRpcRequestProcessor

### CODE REVIEW

File: components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        whenever(verificationSandboxService.get(cordaHoldingIdentity, cpkSummaries)).thenReturn(sandbox)
+        whenever(sandbox.virtualNodeContext).thenReturn(virtualNodeContext)
+        whenever(virtualNodeContext.holdingIdentity).thenReturn(cordaHoldingIdentity)
+        whenever(currentSandboxGroupContext.get()).thenReturn(sandbox)
+    }
+
+    @Test
+    fun `successful response messages`() {
+        val request = createRequest("r1")
+        val responseRecord = Record("", "1", "")
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt]:
```suggestion
        val responseRecord = Record("", "1", flowEvent)
```

File: components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        whenever(verificationRequestHandler.handleRequest(sandbox, request)).thenReturn(responseRecord)
+
+        val results = verificationRpcRequestProcessor.process(request)
+
+        assertThat(results).isNotNull
+        assertThat(results).isEqualTo(flowEvent)
+        assertThat(results).isEqualTo(responseRecord)
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt]:
```suggestion
        assertThat(results).isEqualTo(responseRecord.value)
```

File: components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        verify(externalEventResponseFactory).success(externalEventContext, response)
```
Comment (mbrkic-r3) [inline components/ledger/ledger-verification/src/test/kotlin/net/corda/ledger/verification/processor/impl/VerificationRpcRequestProcessorTest.kt]: This won't work as `externalEventResponseFactory` is accessed in `verificationRequestHandler.handleRequest` and that is mocked (`whenever(verificationRequestHandler.handleRequest(sandbox, request)).thenReturn(responseRecord)`)

**Review by mbrkic-r3 — APPROVED** [review]
LGTM

**Review by lankydan — APPROVED** [review]

**Pull request merged**